core-rust/state-manager/src/types.rs

```diff
@@ -200,6 +202,54 @@ impl HasLedgerPayloadHash for NotarizedTransaction {
     }
 }
 
+#[derive(
+    PartialEq,
+    Eq,
+    Hash,
+    Clone,
+    Copy,
+    PartialOrd,
+    Ord,
+    ScryptoCategorize,
+    ScryptoEncode,
+    ScryptoDecode,
+)]
+pub struct LedgerReceiptHash([u8; Self::LENGTH]);
+
+impl LedgerReceiptHash {
+    pub const LENGTH: usize = 32;
+
+    pub fn for_receipt(receipt: &LedgerTransactionReceipt) -> Self {
+        Self::for_receipt_bytes(&scrypto_encode(receipt).unwrap())
```

Comment:
Note from author:

It turns out that the current (very direct) hash is not what we want.
As pointed out by @iamyulong, we want to exclude some 🟧 things from merklization:

![image](https://user-images.githubusercontent.com/119413677/222091518-1819e210-4023-4978-a420-ab184fae965d.png)
([source](https://rdxworks.slack.com/archives/C01M90Y2448/p1677660105653719?thread_ts=1677656078.643539&cid=C01M90Y2448))

Reply:
in my interpretation, this means that we do NOT want to hash `LedgerTransactionReceipt`'s:
- `fee_summary` (mentioned explicitly on the whiteboard)
- `application_logs` (mentioned explicitly; we don't have a way to filter for "system logs", though)
- `resource_changes` (that's how I interpret the mentioned "resource movement")
- `entity_changes` (not mentioned by the whiteboard, but very similar to `resource_changes` ❓ )
- `next_epoch` (the whiteboard mentions "epoch+round", so this one too ❓ )

This would mean that `LedgerTransactionReceipt` would only hash:
- `outcome` (obviously)
- `substate_changes` (yeah they are important, but our independently calculated substate root hash kinda captures that anyways ❓ )

@iamyulong you said that the whiteboard may be a little outdated; please advice on the "❓" above.

@LukasGasior1 I am curious what you think here.

Reply:
_(I want to resolve it here, since I believe the change belongs in this very PR)_

Reply:
My interpretation:
- `entity_changes`: No, can be derived from `entity_changes`
- `next_epoch`: No, will be covered by events (in follow-up PRs)
- `substate_changes`: Yes
- `outcome`: Suggest we start with mapping it to a `bool` for sucess or failure (the error code not designed to be merkelizable)

Reply:
@jakrawcz-rdx yeah, sounds good. The original plan was to split the receipt into "LedgerReceipt" and "ExtendedReceipt", where LedgerReceipt is what validators agree on (and all nodes store it), while ExtendedReceipt would be a container for all other info about the transaction that the node decides to keep (for querying purposes). But I guess that's not something we want to introduce in this PR

Reply:
Thanks for the clarification, PTAL at the latest commit 👍

Reply:
that'll do for now :)